### PR TITLE
greenkeeper/initial

### PR DESCRIPTION
**Pull request by NeoPhi**
This adds greenkeeper and does an update all of dependencies used by the project. This included some TypeScript and tslint configuration changes as part of the version upgrade.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,27 +18,27 @@
     "pretest": "npm run compile",
     "test": "npm run testonly --",
     "posttest": "npm run lint",
-    "lint": "tslint ./src/**/*.ts",
+    "lint": "tslint --type-check --project ./tsconfig.json ./src/**/*.ts",
```
Comment (helfer) [inline package.json]: Just out of curiosity, why are these options necessary?

File: package.json [hunk 2]
```diff
@@ -18,27 +18,27 @@
...
     "watch": "tsc -w",
     "testonly": "mocha --reporter spec --full-trace ./dist/test/tests.js",
     "coverage": "node ./node_modules/istanbul/lib/cli.js cover _mocha -- --full-trace ./dist/test/tests.js",
     "postcoverage": "remap-istanbul --input coverage/coverage.raw.json --type lcovonly --output coverage/lcov.info"
   },
   "devDependencies": {
+    "@types/graphql": "^0.9.0",
```
Comment (helfer) [inline package.json]: This probably means we'll need to update it in all other related packages, if we haven't already.

File: src/test/tests.ts [hunk 3]
```diff
@@ -1,3 +1,6 @@
+// chai style expect().to.be.true violates no-unused-expression
+/* tslint:disable:no-unused-expression */
```
Comment (helfer) [inline src/test/tests.ts]: Normally I'd argue for applying this line by line only, but there are probably lots of these sprinkled throughout the file, so this is okay.

File: package.json (hunk 1 above)
Comment (NeoPhi) [inline package.json]: The latest version of tslint issues the following warnings without those command line flags set:
```
Warning: The 'no-unused-variable' rule requires type checking
Warning: The 'no-use-before-declare' rule requires type checking
```

**Pull request merged**